### DISCUSSION TITLE
USB Power event to then use HDMI out?

### DISCUSSION THREAD

**Discussion by NicK4rT** [gopro/labs#714] [Q&A] [answered]
Hi there,

I am currently trying to set up a GoPro (Hero 11 black) to start upon USB connection, go into HDMI mode and shut down upon USB disconnect.

The GoPro is using MediaMod and will have an HDMI connected to it at all times.

I have gotten as far as generating codes for the USB power events, which works, however the GoPro does not pass HDMI through the HDMI cable and I have no clue with which command I can get it to do so. Any help would be appreciated!

**Comment by NicK4rT** [comment]
To partially answer own question: 

Using !uN!S!u10E!R did NOT work, so I have now ended up at !MWAKE=2<u0!1O!R, which does work, at least partially, as it does not shut down once USB is disconnected. The Wake part works repeatedly though.
I would like this to run pretty much forever. HDMI, is manually configured with o!HDMI=2

**Reply by NicK4rT** [reply]
Have gotten to !MWAKE=2!MBOOT="!Lbt"!SAVEbt=<u0!1O!R which still does not work. To be continued!

**Reply by NicK4rT** [reply]
Shoutout to blocky coding which helps: https://gopro.github.io/labs/build/

**Reply by NicK4rT** [reply]
Problem with the !MWAKE command seems that I can invert it accidentally (such that startup happens on USB disconnect instead of connect)

**Reply by dnewman-gpsw** [reply, marked as answer]
!MHDMI=1!MWAKE=2!MBOOT="!Lbt"!SAVEbt=!1N<u0!1O!R

This is a little friendlier, adding the !1N so the camera is not testing USB dozens of times a second.  

The false boot should be too much of an issue, and the camera will shutdown with no USB power and the script.

**Reply by NicK4rT** [reply]
Thanks for the input @dnewman-gpsw! :)
I have tried to run this code instead, however, this still results in the GoPro not shutting down on USB disconnect.

Could the lack of SD card have something to do with it?

EDIT: Trying with an SD inserted makes this work. I take it the boot commands are saved on the SD? Is there a way to make this work without SD card?

**Reply by dnewman-gpsw** [reply]
There is, but it is risker.  Writing commands to the SD card means if you make an error, or simply change you mind how you want to use the camera, ejecting the SD card restores the camera.  Without the SD Card (you have been warned) the command is !MHDMI=1!MWAKE=2!MBOOT="!1N<u0!1O!R"

**Reply by NicK4rT** [reply]
Thank you!

Just for making sure: This is something a factory reset of the camera would still be able to delete?

Or are you referring to potentially pushing commands that brick or lock you out of the camera (like in this case you would not be able to start the camera without it being connected to USB)?

---

On a somewhat related note, the whole idea of being able to program and command goPros via QR and scripts is awesome and very inspiring! Thank you for making this so accessible!

**Reply by dnewman-gpsw** [reply]
Correct, you can brick the camera with a command (like this) that forces are reboot and repeats. This is no way to interrupt a boot command that is not on the SD card.  The '1' in !1O gives you one second opportunity to cancel. So this command has an out (other then USB.)  But it is all a bit dangerous.  Why not find some older small SD cards.

**Reply by NicK4rT** [reply]
I agree on the danger part and I've already put in the request for some small SD cards. ;)

The reason for trying it without SD first was the area where the cameras will be used being confidential and having cameras, let alone easily removable cameras with inserted tiny SD cards, makes some people a bit nervous.

Thanks again for the help!
